### DISCUSSION TITLE
How the android app access /sdcard ?

### DISCUSSION THREAD

**Discussion by ccinsz** [beeware/briefcase#1328] [Q&A]
Hi, I want to access the file of /sdcard, but it raises PermissionError: [Errno 13] Permission denied. How can I solve this problem?

--------- beginning of system
06-20 18:34:15.043 28169 28169 D VRI[MainActivity]: mPopUpViewOffsets: offset=(0.0, 0.0), scale=(1.0, 1.0)
06-20 18:34:15.043 28169 28191 I AdrenoGLES-0: QUALCOMM build                   : eaa701cd06, I183f6321f0
06-20 18:34:15.043 28169 28191 I AdrenoGLES-0: Build Date                       : 10/27/22
06-20 18:34:15.043 28169 28191 I AdrenoGLES-0: OpenGL ES Shader Compiler Version: EV031.35.01.12
06-20 18:34:15.043 28169 28191 I AdrenoGLES-0: Local Branch                     :
06-20 18:34:15.043 28169 28191 I AdrenoGLES-0: Remote Branch                    :
06-20 18:34:15.043 28169 28191 I AdrenoGLES-0: Remote Branch                    :
06-20 18:34:15.043 28169 28191 I AdrenoGLES-0: Reconstruct Branch               :
06-20 18:34:15.043 28169 28191 I AdrenoGLES-0: Build Config                     : S P 10.0.7 AArch64
06-20 18:34:15.043 28169 28191 I AdrenoGLES-0: Driver Path                      : /vendor/lib64/egl/libGLESv2_adreno.so
06-20 18:34:15.047 28169 28191 I AdrenoGLES-0: PFP: 0x016dc094, ME: 0x00000000
06-20 18:34:15.095 28169 28191 W Parcel  : Expecting binder but got null!
06-20 18:34:15.104 28169 28169 D MainActivity: onPrepareOptionsMenu() start
06-20 18:34:15.108 28169 28169 D MainActivity: onPrepareOptionsMenu() complete

06-20 18:34:19.219 28169 28169 W python.stderr: Error in async handler: [Errno 13] Permission denied: '/sdcard/tmp/44.txt'
06-20 18:34:19.219 28169 28169 W python.stderr: Traceback (most recent call last):
06-20 18:34:19.219 28169 28169 W python.stderr:   File "/data/data/com.example.demo0529/files/chaquopy/AssetFinder/requirements/toga/handlers.py", line 28, in handler_with_cleanup
06-20 18:34:19.219 28169 28169 W python.stderr:   File "/data/data/com.example.demo0529/files/chaquopy/AssetFinder/app/demo0529/app.py", line 72, in mytest
06-20 18:34:19.220 28169 28169 W python.stderr: PermissionError: [Errno 13] Permission denied: '/sdcard/tmp/44.txt'

**Comment by freakboy3742** [comment]
At the very least, you will need to request permission to access external files by modifying your Android Gradle project.

File access on mobile devices isn't a simple topic; you can't always "just open a file". Android in particular requires the use of specific APIs to access some file locations. We eventually aim to provide APIs to simplify this process, but for the moment, you'll need to become familiar with [Android's platform documentation for file access](https://developer.android.com/guide/topics/data), and follow those guidelines.

**Reply by mhsmith** [reply]
See https://github.com/beeware/toga/discussions/1990 for recent discussion.

**Reply by ccinsz** [reply]
Thank you for your help.
